### PR TITLE
Creating json

### PR DESCRIPTION
**Pull request by r0nl**
Creating the json file for representing the website network structure.

## Description
Added a json file that stores the network as key-value pair with the current link node being the key and the list of all child link nodes being the value.
This will further help in visualizing the structure and to plot charts and graphs for the same.

## Motivation and Context
The change was required and the current state of representation showed no hierarchy in the structure and was not able to handle redundant links. The changes will enable the crawler to avoid the redundant links and avoid infinite loops.
Open issue solved: https://github.com/PROxZIMA/DarkSpider/issues/1

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: torcrawl.py [hunk 1]
```diff
@@ -274,7 +274,7 @@ def main():
         lst = crawler.crawl()
         with open(out_path + "/links.txt", "w+", encoding="UTF-8") as file:
             for item in lst:
-                file.write(f"{item}\n")
+                file.write(f"{item} \n")
```
Comment (PROxZIMA) [inline torcrawl.py:277]: Is this change necessary? From my point of view this extra white-space is not required.

File: modules/crawler.py [hunk 2]
```diff
@@ -155,15 +177,16 @@ def crawl(self):
                     sys.stdout.write("-- Results: " + str(len(ord_lst)) + "\r")
                     sys.stdout.flush()
 
-                # Pause time.
-                if float(self.c_pause) > 0:
```
Comment (PROxZIMA) [inline modules/crawler.py:159]: Keep this condition as it is foolproof.

**Review by PROxZIMA — CHANGES_REQUESTED** [review]

File: torcrawl.py (hunk 1 above)
Comment (r0nl) [reply torcrawl.py:277]: Oh I did it for making the manual search in links.txt easier. So that I can analyze and verify results. You can remove the extra whitespace.

File: modules/crawler.py (hunk 2 above)
Comment (r0nl) [reply modules/crawler.py:159]: c_pause has a default value of 1, hence the if statement felt unnecessary.

**Review by PROxZIMA — APPROVED** [review]

File: modules/crawler.py [hunk 3]
```diff
@@ -130,6 +141,17 @@ def crawl(self):
...
+                        #external links overwriting for rerun
+                        with open(self.out_path + "/extlinks.txt", "w+", encoding="UTF-8") as lst_file:
+                            lst_file.write('')
+                        #telephone numbers overwriting for rerun
+                        with open(self.out_path + "/telephones.txt", "w+", encoding="UTF-8") as lst_file:
+                            lst_file.write('')
+                        #mails overwriting for rerun
+                        with open(self.out_path + "/mails.txt", "w+", encoding="UTF-8") as lst_file:
+                            lst_file.write('')
+
```
Comment (PROxZIMA) [inline modules/crawler.py:154]: Why is this written here? `self.excludes(link)` will do this job right?

File: modules/crawler.py [hunk 4]
```diff
@@ -155,15 +177,16 @@ def crawl(self):
...
-                    it_code = html_page.getcode()
-                    with open(log_path, "w+", encoding="UTF-8") as log_file:
-                        log_file.write(f"[{str(it_code)}] {str(item)} \n")
+                # Adding to json data
+                cur_level_prev_step = cur_level.difference(ord_lst_clone)
+                ord_lst_clone =  cur_level.union(ord_lst_clone)
+                if item[-1]== '/':
+                    item = item[:-1]
+                if item not in json_data.keys():
+                    json_data[item] = list(cur_level_prev_step)
```
Comment (PROxZIMA) [inline modules/crawler.py:189]: This and next few lines needs to be optimized too.
